pkg/endpoint/bpf.go

```diff
@@ -56,7 +56,10 @@ const (
 	ExecTimeout = 300 * time.Second
 
 	// EndpointGenerationTimeout specifies timeout for proxy completion context
-	EndpointGenerationTimeout = 55 * time.Second
+	// Must be greater than `ExecTimeout` as a context with this timeout
+	// is started before using `ExecTimeout` and still used after `ExecTimeout`
+	// is already canceled.
+	EndpointGenerationTimeout = 330 * time.Second
```

Comment:
This is a tough one, if something takes more than 330 seconds to build it's just straight wrong. We can't expect that an endpoint took more than 20 seconds to enforce a policy. I know there are hardware constraints but we should get reasonable here in order to enforce a maximum acceptable value.

Reply:
@aanm See the note https://github.com/cilium/cilium/pull/6211#issuecomment-437300178 above. It is quite common that the sidecar proxy container within an Istio pod may take 2-3 minutes to start up. If we time out before the sidecar proxy gets and acks the L7 network policy the endpoint ends up in `not-ready` state, which may linger indefinitely.

Reply:
The point of this PR is to make the longer timeouts less harmful, by allowing other endpoints to regenerate as soon as the resource intensive parts of regeneration are done.

We already do return from the API call on the initial build without waiting for proxy completions. If I recall correctly the purpose of that is to allow the pod initialization to continue after L3/4 networking is up and running (@ianvernon correct if I'm wrong here!).

Reply:
We also have seen actual bpf compilations take longer than a minute in the past on very resource constrained systems. This has improved a lot (and is still improving) for the typical case, but IMO the worst case scenario still exists?

Either way, I'd leave fine tuning the timeout for a separate PR.